versioned_docs/version-4.6/developers/applications/data-loader.md

```diff
@@ -4,178 +4,122 @@ title: Data Loader
 		{
 			"id": 1,
-			"username": "admin",
-			"email": "admin@example.com",
-			"role": "administrator"
+			"name": "Harper",
+			"breed": "Labrador",
+			"age": 3,
+			"tricks": ["sit"]
 		},
 		{
 			"id": 2,
-			"username": "user1",
-			"email": "user1@example.com",
-			"role": "standard"
+			"name": "Balto",
+			"breed": "Husky",
+			"age": 5,
+			"tricks": ["run", "pull sled"]
 		}
 	]
 }
 ```
 
-### Multiple Tables
+This file tells Harper: _“Insert these two records into the `Dog` table when this app runs.”_
 
-To load data into multiple tables, create separate data files for each table:
+## Connecting the Data Loader
 
-**users.json:**
-
-```json
-{
-	"database": "myapp",
-	"table": "users",
-	"records": [
-		{
-			"id": 1,
-			"username": "admin",
-			"email": "admin@example.com"
-		}
-	]
-}
-```
-
-**settings.yaml:**
-
-```yaml
-database: myapp
-table: settings
-records:
-  - id: 1
-    setting_name: app_name
-    setting_value: My Application
-  - id: 2
-    setting_name: version
-    setting_value: '1.0.0'
-```
-
-## File Organization
-
-You can organize your data files in various ways:
-
-### Single File Pattern
+Next, let’s tell Harper to use this file when running the application. Open `config.yaml` in the root of your project and add:
 
 ```yaml
 dataLoader:
-  files: 'data/seed-data.json'
+  files: 'data/dogs.json'
 ```
 
-### Multiple Files Pattern
+That’s it. Now the Data Loader knows where to look.
 
-```yaml
-dataLoader:
-  files:
-    - 'data/users.json'
-    - 'data/settings.yaml'
-    - 'data/initial-products.json'
-```
+## Running with Data
 
-### Glob Pattern
+Go ahead and start your app again:
 
-```yaml
-dataLoader:
-  files: 'data/**/*.{json,yaml,yml}'
+```bash
+harperdb dev .
 ```
 
-## Loading Behavior
+This time, when Harper runs, it will automatically read `dogs.json` and load the records into the Dog table. You don’t need to write any import scripts or SQL statements, it just works.
 
-When Harper starts up with a component that includes the Data Loader:
+You can confirm the data is there by hitting the endpoint you created earlier:
 
-1. The Data Loader reads all specified data files (JSON or YAML)
-1. For each file, it validates that a single table is specified
-1. Records are inserted or updated based on timestamp comparison:
-   - New records are inserted if they don't exist
-   - Existing records are updated only if the data file's modification time is newer than the record's updated time
-   - This ensures data files can be safely reloaded without overwriting newer changes
-1. If records with the same primary key already exist, updates occur only when the file is newer
-
-Note: While the Data Loader can create tables automatically by inferring the schema from the provided records, it's recommended to define your table schemas explicitly using the [graphqlSchema](../applications/defining-schemas) component for better control and type safety.
-
-## Best Practices
+```bash
+curl http://localhost:9926/Dog/
+```
```

Comment:
Its a subtle detail but the trailing `/` is important here and some devs might miss that. Lets add a note that calls attention to this and then links to the REST docs for more information on how to query tables/resources/records using the REST plugin.